### DISCUSSION TITLE
Module Definitions Translation tab?

### DISCUSSION THREAD

**Discussion by leigh-pointer** [oqtane/oqtane.framework#2321] [Q&A]
Should that not list the translations for the Module ? if yes, are we permitted to add module translations to the Registry ?

**Comment by sbwalker** [comment]
Yes, the new Translations tab in the Edit Module Definition is integrated with the Registry. It is important to note that the PackageName property in the IModule interface is used as the key for identifying module translation packages in the Registry. So if your custom module is not currently setting the PackageName property, it will need to. In regards to the Registry, you should set the Product Type to Translation... and then provide the PackageId using the same prefix as implemented in the IModule interface PackageName property. This will provide the association which will allow the translation to be displayed in the Edit Module Definition user interface.

**Reply by JanOlsmar** [reply]
Shouldn't there be an option to upload a translation to that module.

**Reply by sbwalker** [reply]
There is in fact only a single installable package format in Oqtane (the standard Nuget package format) so regardless of whether you are installing a module, a theme, or a translation they are all using the same exact package format. So in reality Oqtane could have a single "package upload" UI for the entire framework where a user could upload modules, themes, translations, etc... it does not matter as they all use the same package format. However to make it simpler for users, an upload option was included in the Add Language, Add Module, Add Theme UI - even though they all do the exact same thing. So to answer your question... a module translation could be uploaded in any of the existing upload UIs - so I am not sure if it makes sense to include yet another upload option in Edit Module?

**Reply by JanOlsmar** [reply]
Maybe. I have translations for my modules. I get a message No translation exist. For other modules I get  a list of irrelevant nuget packages.
I would like to find on my modules the version of translation installed.

**Reply by sbwalker** [reply]
Ok, I see your point... currently the new Translation tab only shows module translations which exist in the Registry - but it would be more useful if it also showed the translations that are already installed (including version). I can make this change.

**Comment by sbwalker** [comment]
@JanOlsmar the module translations tab has been refactored in #2325 to display a list which is a combination of the module translations which have already been installed (including version#) as well as any additional module translations which may exist in the Registry. Please note (as mentioned previously) - the PackageName property on the module's IModule interface must be set in order for this functionality to work properly. If a module has not set the PackageName property, this tab will not display any results.

![image](https://user-images.githubusercontent.com/4840590/182247379-78e50840-1827-4686-bdf0-67d7cdd2dd9c.png)

**Reply by JanOlsmar** [reply]
Looks perfect
